sample/ema_throughput.go

```diff
@@ -78,6 +81,8 @@ func (d *EMAThroughputSampler) Start() error {
 func (d *EMAThroughputSampler) GetSampleRate(trace *types.Trace) (rate uint, keep bool, reason string, key string) {
 	key = d.key.build(trace)
 	count := int(trace.DescendantCount())
+	// update the dynsampler with the latest goal throughput in case cluster size changed
+	d.dynsampler.GoalThroughputPerSec = d.goalThroughputPerSec / d.GetClusterSize(d.useClusterSize)
```

Comment:
Looking through the emathroughput sampler in dynsampler I don't see where `GoalThroughputPerSec` is used within `GetSampleRate`.  Whats happening that requires us to update `GoalThroughputPerSec` here?

Reply:
It's not used directly within GetSampleRate, it's used when the rates are recalculated after every AdjustmentInterval. But I hadn't thought of a way to not do it on every call to GetSampleRate. 

However, now that you point this out, I think I could instead register a callback that would adjust it and we wouldn't have to set it on every call. I'm going to try that and see how it feels.

Reply:
I've done it this way and I think it's a little cleaner. Please look again.

Reply:
Ya I like this a lot